### ISSUE TITLE
Syntax error in alsa-capabilities on OS X 10.10 Yosemite prevents script from running

### ISSUE THREAD

**Issue by kvcrawford** [ronalde/mpd-configure#10]
Output is the same regardless of the `DEBUG` setting. All other configurations are commented out.

```
$ bash mpd-configure > mpd.conf
./alsa-capabilities: line 500: syntax error near unexpected token `;;'
./alsa-capabilities: line 500: `		;;'

Error: required script `./alsa-capabilities' not found
```

**Comment by ronalde** [comment]
Thanks kvcrawford for your remark. 

After analyzing your question, it seems that my script(s) are incompatible with the version of bash on OSX (version 3.2, while my testing environments on Debian and Arch contain the 4.x versions). See below for technical details. 

However, even though the bash incompatibilties could be resolved relatively easy, it wouldn't bring you any good, because the script depends on the "Advanced Linux Sound Architecture" (alsa). As the name suggests, that's a no go for OSX (or Windows) and depends on (linux) kernel drivers and such. 

What you could try, is to [grab a live linux-image, boot your mac from it](http://www.ubuntu.com/download/desktop/create-a-usb-stick-on-mac-osx) and then [download and run the script](http://lacocina.nl/detect-alsa-output-capabilities), to see the information for your sound hardware. I'm not sure however, if alsa has kernel support for sound hardware on Apple hardware;  models equipped with Intel's HD Audio chipsets should work. 

If you would try this, I would be very interested in the results.

Sorry for any inconvenience and the delay for answering your question.

Regards,
Ronald

## Details of the bash incompatibilities

The specific problem `syntax error near unexpected token ``;;'` was easily resolved by removing the double semicolumns (`... ;;`) at the end of the last item in a (select) case statement, see commit 47457998b4f3321a694be51614803d0ff65ebbf2. This makes the code a bit more portable. 

But bash 3.2 does not support associative arrays (`declare -A SOMEARRAY`), which the script uses extensively. Those could be replaced with simpler indexed arrays (`.. declare -a SOMEARRAY`) for storing the keys, while combining this with an extra (indexed) array to hold the values associated with those keys. See http://wiki.bash-hackers.org/syntax/arrays for extra clarification.

**Comment by ronalde** [comment]
As a side note; if your tying to achieve audio nirvana, I'm sure that is possible using your native Apple platform and software. [MEASUREMENTS: Bit-Perfect Audiophile Music Players (Mac OS X)] (http://archimago.blogspot.nl/2013/05/measurements-bit-perfect-audiophile.html) seems to  be a good place to start.

**Comment by kvcrawford** [comment]
Wow, what an in-depth response. Thank you!

I suspected that the script was simply incompatible with OS X. If I had just looked up what ALSA stands for then I would have had my answer. D'oh!

Given the results of the link you posted, I think it's safe to say that default configuration with CoreAudio is already bit perfect. He said that all the players are bit perfect, and only mentions one (Audirvana Plus) that can bypass CoreAudio.

Thanks again!
